Mathlib/GroupTheory/Index.lean

```diff
@@ -521,16 +513,41 @@ class _root_.AddSubgroup.FiniteIndex {G : Type*} [AddGroup G] (H : AddSubgroup G
 @[deprecated (since := "2025-04-13")]
 alias _root_AddSubgroup.FiniteIndex.finiteIndex := AddSubgroup.FiniteIndex.index_ne_zero
 
+variable (H) in
+/-- Typeclass for finite index subgroups. -/
+@[to_additive] class FiniteIndex : Prop where
+  /-- The subgroup has finite index;
+  recall that `Subgroup.index` returns 0 when the index is infinite. -/
+  index_ne_zero : H.index ≠ 0
+
 @[deprecated (since := "2025-04-13")] alias FiniteIndex.finiteIndex := FiniteIndex.index_ne_zero
 
+/-- Typeclass for a subgroup `H` to have finite index in a subgroup `K`. -/
+class _root_.AddSubgroup.IsFiniteRelIndex {G : Type*} [AddGroup G] (H K : AddSubgroup G) :
+    Prop where
+  protected relIndex_ne_zero : H.relindex K ≠ 0
```

Comment:
Since the definition name is currently `relindex`, the field should be called `relindex_ne_zero` (and similarly below).